internal/service/ecs/task_definition.go

```diff
@@ -143,6 +143,29 @@ func ResourceTaskDefinition() *schema.Resource {
 				ValidateFunc: validation.StringInSlice(ecs.NetworkMode_Values(), false),
 			},
 
+			"runtime_platform": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"operating_system_family": {
+							Type:         schema.TypeString,
+							Required:     true,
```

Comment:
According to https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_RuntimePlatform.html non are required

Reply:
The other doc for the definitions said it was conditionally required if fargate was enabled 🤦‍♂️

Reply:
ooi what's the difference between `Required: false` and `Optional: true`?

Reply:
There isnt AFAIK. the convention is not to use `Required: false` but rather be explicit with `Optional: true`

Reply:
```
	// If one of these is set, then this item can come from the configuration.
	// Both cannot be set. If Optional is set, the value is optional. If
	// Required is set, the value is required.
	//
	// One of these must be set if the value is not computed. That is:
	// value either comes from the config, is computed, or is both.
	Optional bool
	Required bool
```

Threw me off; don't see many things like that in .net